modules/@apostrophecms/asset/index.js

```diff
@@ -22,12 +23,21 @@ module.exports = {
       ...globalIcons
     };
   },
+  handlers (self) {
+    return {
+      'apostrophe:modulesReady': {
+        async runUiBuildTask() {
+          await self.apos.task.invoke('@apostrophecms/asset:build');
+        }
+      }
+    };
+  },
   tasks(self) {
     return {
       build: {
         usage: 'Build Apostrophe frontend javascript master import files',
         afterModuleInit: true,
-        async task(argv) {
+        async task() {
```

Comment:
If someone invokes the task as a task, it should do all the things by default. I think people expect that.

Whereas if the handler invokes it, it should pass an option specifying the "only if you think it's necessary" behavior.

Reply:
Also modulesReady should not invoke the task if `self.apos.isTask()` is true. We don't want asset builds happening during, say, migrations. And we don't want a conditional asset build happening before a full asset build in the same invocation either if the command line task requested is the actual asset build task.